### PR TITLE
Write ReplicaSet Replace and Patch Test +2 Endpoints

### LINKED ISSUE #99134: Write ReplicaSet Replace and Patch Test +2 Endpoints
# Progress <code>[4/5]</code>

-   [X] APISnoop org-flow : [ReplicaSetScaleTest.org](https://github.com/cncf/apisnoop/blob/master/tickets/k8s/)
-   [X] test approval issue : [kubernetes/kubernetes#99134](https://github.com/kubernetes/kubernetes/issues/99134)
-   [x] test pr : kuberenetes/kubernetes[#99380](https://github.com/kubernetes/kubernetes/pull/99380)
-   [x] two weeks soak start date : [Link](https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default&width=5&include-filter-by-regex=ReplicaSet%20Replace%20and%20Patch%20tests)
-   [ ] two weeks soak end date : 8 March 2021
-   [ ] test promotion pr : kubernetes/kubernetes#?

# Identifying an untested feature Using APISnoop

According to this APIsnoop query, there are still some remaining Replicaset endpoints that are untested.

with this query, you can filter untested endpoints by their category and eligibility for conformance. e.g below shows a query to find all conformance eligible untested, stable, core endpoints

```sql-mode
SELECT
  endpoint,
  -- k8s_action,
  -- path,
  -- description,
  kind
  FROM testing.untested_stable_endpoint
  where eligible is true
  and endpoint like '%ReplicaSet%'
    and endpoint not like '%Status%'
  --and category = 'core'
  order by kind, endpoint desc
  limit 25;
```

```example
                  endpoint                  |    kind
--------------------------------------------|------------
 patchAppsV1NamespacedReplicaSet            | ReplicaSet
 listAppsV1ReplicaSetForAllNamespaces       | ReplicaSet
 deleteAppsV1CollectionNamespacedReplicaSet | ReplicaSet
(3 rows)

```
replaceAppsV1NamespacedReplicaSet  does not show up as untested as it touched by another [e2e test](https://github.com/ii/kubernetes/commit/d4b38853a591880c69c283268e580ccb46247ee5) not promoted to Conformance.

# API Reference and feature documentation

-   [Kubernetes API Reference Docs](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#-strong-write-operations-replicaset-v1-apps-strong-)
-   [client-go - ReplicaSet](https://pkg.go.dev/k8s.io/kubernetes/test/e2e/framework/replicaset)

# The mock test

## Test outline

1.  Create a ReplicaSet with a static label

2.  Read the ReplicaSet info

3.  Scale the ReplicaSets to 2

4.  Patch the ReplicaSet

5.  Read the ReplicaSet to ensure it is Patch

6.  Delete the Namespace and ReplicaSet

### Test the functionality in Go

Due to the complexity of setting up the resources for APIService we have used the current e2e test as template. It has been extended in a [new ginkgo test](https://github.com/ii/kubernetes/commit/d4b38853a591880c69c283268e580ccb46247ee5) for review.

# Verifying increase in coverage with APISnoop

```sql-mode
select distinct  endpoint, useragent
                 -- to_char(to_timestamp(release_date::bigint), ' HH:MI') as time
from testing.audit_event
where endpoint ilike '%ReplicaSet%'
and useragent ilike '%ReplicaSet lifecycle tests%'
-- and release_date::BIGINT > round(((EXTRACT(EPOCH FROM NOW()))::numeric)*1000,0) - 60000
-- and useragent like 'kubectl%'
order by endpoint
limit 100;

```

```example
               endpoint               |                                              useragent
--------------------------------------|------------------------------------------------------------------------------------------------------
 createAppsV1NamespacedReplicaSet     | e2e.test/v0.0.0 (linux/amd64) kubernetes/$Format -- [sig-apps] ReplicaSet ReplicaSet lifecycle tests
 patchAppsV1NamespacedReplicaSet      | e2e.test/v0.0.0 (linux/amd64) kubernetes/$Format -- [sig-apps] ReplicaSet ReplicaSet lifecycle tests
 readAppsV1NamespacedReplicaSet       | e2e.test/v0.0.0 (linux/amd64) kubernetes/$Format -- [sig-apps] ReplicaSet ReplicaSet lifecycle tests
 replaceAppsV1NamespacedReplicaSet    | e2e.test/v0.0.0 (linux/amd64) kubernetes/$Format -- [sig-apps] ReplicaSet ReplicaSet lifecycle tests
(4 rows)

```

# Final notes

If a test with these calls gets merged, ****test coverage will go up by 2 points****

This test is also created with the goal of conformance promotion.

---

/sig testing
/sig architecture
/area conformance
/sig apps

### PR DESCRIPTION
**Pull request by riaankleinhans**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it:**
This PR adds a test to test the following untested endpoints:
- replaceAppsV1NamespacedReplicaSet
- patchAppsV1NamespacedReplicaSet

**Which issue(s) this PR fixes:**
Fixes #99134

**Testgrid Link:**
[Link](https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default&width=5&include-filter-by-regex=ReplicaSet%20Replace%20and%20Patch%20tests)

**Special notes for your reviewer:**
Adds +2 endpoint test coverage (good for conformance)

**Does this PR introduce a user-facing change?:**
```
NONE

```

**Release note:**
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:**
```
NONE

```

/sig testing
/sig architecture
/sig apps
/area conformance

### CODE REVIEW

**Comment by Riaankl** [comment]
/test pull-kubernetes-e2e-kind

Unrelated tests failed
```
Kubernetes e2e suite: [sig-storage] PersistentVolumes-local [Volume type: tmpfs] Two pods mounting a local volume at the same time should be able to write from pod1 and read from pod2 expand_more5m37s | Kubernetes e2e suite: [sig-storage] PersistentVolumes-local [Volume type: tmpfs] Two pods mounting a local volume at the same time should be able to write from pod1 and read from pod2 expand_more | 5m37s
-- | -- | --
Kubernetes e2e suite: [sig-storage] PersistentVolumes-local [Volume type: tmpfs] Two pods mounting a local volume at the same time should be able to write from pod1 and read from pod2 expand_more | 5m37s
Kubernetes e2e suite: [sig-cli] Kubectl client Simple pod should return command exit codes expand_more4m18s | Kubernetes e2e suite: [sig-cli] Kubectl client Simple pod should return command exit codes expand_more | 4m18s
Kubernetes e2e suite: [sig-cli] Kubectl client Simple pod should return command exit codes expand_more | 4m18s
Kubernetes e2e suite: [sig-storage] CSI mock volume CSI Volume expansion should not expand volume if resizingOnDriver=off, resizingOnSC=on expand_more5m36s | Kubernetes e2e suite: [sig-storage] CSI mock volume CSI Volume expansion should not expand volume if resizingOnDriver=off, resizingOnSC=on expand_more | 5m36s
Kubernetes e2e suite: [sig-storage] CSI mock volume CSI Volume expansion should not expand volume if resizingOnDriver=off, resizingOnSC=on expand_more | 5m36s
Kubernetes e2e suite: [sig-network] Services should be able to change the type from ClusterIP to ExternalName [Conformance] expand_more5m5s | Kubernetes e2e suite: [sig-network] Services should be able to change the type from ClusterIP to ExternalName [Conformance] expand_more | 5m5s
Kubernetes e2e suite: [sig-network] Services should be able to change the type from ClusterIP to ExternalName [Conformance] expand_more | 5m5s
Kubernetes e2e suite: [sig-apps] Deployment iterative rollouts should eventually progress expand_more5m49s | Kubernetes e2e suite: [sig-apps] Deployment iterative rollouts should eventually progress expand_more | 5m49s
Kubernetes e2e suite: [sig-apps] Deployment iterative rollouts should eventually progress expand_more | 5m49s
Kubernetes e2e suite: [sig-network] KubeProxy should set TCP CLOSE_WAIT timeout [Privileged] expand_more | Kubernetes e2e suite: [sig-network] KubeProxy should set TCP CLOSE_WAIT timeout [Privileged] expand_more
Kubernetes e2e suite: [sig-network] KubeProxy should set TCP CLOSE_WAIT timeout [Privileged] expand_more
```

File: test/e2e/apps/replica_set.go [hunk 1]
```diff
@@ -412,3 +416,63 @@ func testRSScaleSubresources(f *framework.Framework) {
...
+	rsPatchImage := imageutils.GetE2EImage(imageutils.Pause)
+
+	// Create a ReplicaSet
+	rs := newRS(rsName, replicas, rsPodLabels, WebserverImageName, WebserverImage, nil)
+	_, err := c.AppsV1().ReplicaSets(ns).Create(context.TODO(), rs, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	// Verify that the required pods have come up.
+	err = e2epod.VerifyPodsRunning(c, ns, "sample-pod", false, replicas)
+	framework.ExpectNoError(err, "error in waiting for pods to come up: %s", err)
```
Comment (dims) [inline test/e2e/apps/replica_set.go:444]: better message? we are logging an error during verification ( we are not waiting per se )

File: test/e2e/apps/replica_set.go [hunk 2]
```diff
@@ -412,3 +416,63 @@ func testRSScaleSubresources(f *framework.Framework) {
...
+
+	// Scale the ReplicaSet
+	ginkgo.By(fmt.Sprintf("Scaling up %q replicaset ", rsName))
+	_, err = e2ereplicaset.UpdateReplicaSetWithRetries(c, ns, rsName, func(update *appsv1.ReplicaSet) {
+		x := int32(2)
+		update.Spec.Replicas = &x
+	})
+	framework.ExpectNoError(err)
```
Comment (dims) [inline test/e2e/apps/replica_set.go:452]: add some detail message? ( not everyone will be looking at matching sources when they are browsing a log )

File: test/e2e/apps/replica_set.go [hunk 3]
```diff
@@ -412,3 +416,63 @@ func testRSScaleSubresources(f *framework.Framework) {
...
+					"containers": [1]map[string]interface{}{{
+						"name":    rsName,
+						"image":   rsPatchImage,
+						"command": []string{"/bin/sleep", "100000"},
+					}},
+				},
+			},
+		},
+	})
+	framework.ExpectNoError(err, "failed to Marshal Deployment JSON patch")
```
Comment (dims) [inline test/e2e/apps/replica_set.go:474]: s/Deployment/ReplicaSet ?

File: test/e2e/apps/replica_set.go [hunk 4]
```diff
@@ -412,3 +416,63 @@ func testRSScaleSubresources(f *framework.Framework) {
...
+	_, err = f.ClientSet.AppsV1().ReplicaSets(ns).Patch(context.TODO(), rsName, types.StrategicMergePatchType, []byte(rsPatch), metav1.PatchOptions{})
+	framework.ExpectNoError(err, "failed to patch ReplicaSet")
+
```
Comment (dims) [inline test/e2e/apps/replica_set.go:477]: - Do we want to verify that the patch applied correctly with a get?
- Do we want to wait (`e2ereplicaset.WaitForReadyReplicaSet`)?

File: test/e2e/apps/replica_set.go (hunk 1 above)
Comment (Riaankl) [reply test/e2e/apps/replica_set.go]: You are correct, Thank you.
Updated with a better message.

File: test/e2e/apps/replica_set.go (hunk 2 above)
Comment (Riaankl) [reply test/e2e/apps/replica_set.go]: Correct, Thank you.
Added an error message.

File: test/e2e/apps/replica_set.go (hunk 3 above)
Comment (Riaankl) [reply test/e2e/apps/replica_set.go]: Good catch, thank you.
Updated to ReplicaSet

File: test/e2e/apps/replica_set.go (hunk 4 above)
Comment (Riaankl) [reply test/e2e/apps/replica_set.go:477]: I do not think SIG Apps want `e2ereplicaset.WaitForReadyReplicaSet` in the tests. Last time I had to remove it.
Your thoughts @soltysh ?

Comment (dims) [reply test/e2e/apps/replica_set.go:477]: @Riaankl we can do the first one at least then?

Comment (Riaankl) [reply test/e2e/apps/replica_set.go:477]: Yes, sorry. Two different issues. A `get` make sense.
Added and tested it.
Thank you.

File: test/e2e/apps/replica_set.go [hunk 5]
```diff
@@ -412,3 +416,66 @@ func testRSScaleSubresources(f *framework.Framework) {
...
+			},
+		},
+	})
+	framework.ExpectNoError(err, "failed to Marshal ReplicaSet JSON patch")
+	_, err = f.ClientSet.AppsV1().ReplicaSets(ns).Patch(context.TODO(), rsName, types.StrategicMergePatchType, []byte(rsPatch), metav1.PatchOptions{})
+	framework.ExpectNoError(err, "failed to patch ReplicaSet")
+
+	rs, err = c.AppsV1().ReplicaSets(ns).Get(context.TODO(), rsName, metav1.GetOptions{})
+	framework.ExpectNoError(err, "Failed to get replicaset resource: %v", err)
+	framework.ExpectEqual(*(rs.Spec.Replicas), int32(3), "replicaset should have 3 replicas")
```
Comment (dims) [inline test/e2e/apps/replica_set.go:480]: use `rsPatchReplicas` instead of `int32(3)`?

also do we want to cross check if `rsPatchImage` was applied too?

Comment (Riaankl) [reply test/e2e/apps/replica_set.go]: Update `rsPatchReplicas` instead of `int32(3)`
Just a little stuck on how to check `rsPatchImage`

Comment (Riaankl) [reply test/e2e/apps/replica_set.go]: Got the check for `rsPatchImage` in and working.
Thanks for the review @dims

**Comment by dims** [comment]
LGTM over to you @soltysh

**Comment by Riaankl** [comment]
/test pull-kubernetes-e2e-gce-ubuntu-containerd

Unrelate `Network` and `Storage` test flakes
```
Kubernetes e2e suite: [sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: Dynamic PV (default fs)] subPath should support file as subpath [LinuxOnly] expand_more7m1s | Kubernetes e2e suite: [sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: Dynamic PV (default fs)] subPath should support file as subpath [LinuxOnly] expand_more | 7m1s
Kubernetes e2e suite: [sig-storage] CSI Volumes [Driver: csi-hostpath] [Testpattern: Dynamic PV (default fs)] subPath should support file as subpath [LinuxOnly] expand_more | 7m1s
Kubernetes e2e suite: [sig-network] Networking Granular Checks: Services should function for multiple endpoint-Services with same selector expand_more2m44s | Kubernetes e2e suite: [sig-network] Networking Granular Checks: Services should function for multiple endpoint-Services with same selector expand_more | 2m44s
Kubernetes e2e suite: [sig-network] Networking Granular Checks: Services should function for multiple endpoint-Services with same selector expand_more | 2m44s
Kubernetes e2e suite: [sig-network] Networking Granular Checks: Services should update endpoints: http expand_more3m2s | Kubernetes e2e suite: [sig-network] Networking Granular Checks: Services should update endpoints: http expand_more | 3m2s
Kubernetes e2e suite: [sig-network] Networking Granular Checks: Services should update endpoints: http expand_more | 3m2s
Kubernetes e2e suite: [sig-network] Services should be able to update service type to NodePort listening on same port number but different protocols expand_more2m49s | Kubernetes e2e suite: [sig-network] Services should be able to update service type to NodePort listening on same port number but different protocols expand_more | 2m49s
Kubernetes e2e suite: [sig-network] Services should be able to update service type to NodePort listening on same port number but different protocols expand_more | 2m49s
Kubernetes e2e suite: [k8s.io] [sig-storage] ConfigMap optional updates should be reflected in volume [NodeConformance] [Conformance] expand_more4m24s | Kubernetes e2e suite: [k8s.io] [sig-storage] ConfigMap optional updates should be reflected in volume [NodeConformance] [Conformance] expand_more | 4m24s
Kubernetes e2e suite: [k8s.io] [sig-storage] ConfigMap optional updates should be reflected in volume [NodeConformance] [Conformance] expand_more | 4m24s
Kubernetes e2e suite: [sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: dir-link] [Testpattern: Pre-provisioned PV (default fs)] subPath should support existing directory expand_more5m34s | Kubernetes e2e suite: [sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: dir-link] [Testpattern: Pre-provisioned PV (default fs)] subPath should support existing directory expand_more | 5m34s
Kubernetes e2e suite: [sig-storage] In-tree Volumes [Driver: local][LocalVolumeType: dir-link] [Testpattern: Pre-provisioned PV (default fs)] subPath should support existing directory expand_more | 5m34s
Kubernetes e2e suite: [sig-storage] In-tree Volumes [Driver: gluster] [Testpattern: Inline-volume (default fs)] subPath should support file as subpath [LinuxOnly] expand_more2m57s | Kubernetes e2e suite: [sig-storage] In-tree Volumes [Driver: gluster] [Testpattern: Inline-volume (default fs)] subPath should support file as subpath [LinuxOnly] expand_more | 2m57s
Kubernetes e2e suite: [sig-storage] In-tree Volumes [Driver: gluster] [Testpattern: Inline-volume (default fs)] subPath should support file as subpath [LinuxOnly] expand_more | 2m57s
e2e.go: Test expand_less22m51serror during ./hack/ginkgo-e2e.sh --ginkgo.skip=\[Slow\]\|\[Serial\]\|\[Disruptive\]\|\[Flaky\]\|\[Feature:.+\] --minStartupPods=8 --report-dir=/workspace/_artifacts --disable-log-dump=true: exit status 1 | e2e.go: Test expand_less | 22m51s | error during ./hack/ginkgo-e2e.sh --ginkgo.skip=\[Slow\]\|\[Serial\]\|\[Disruptive\]\|\[Flaky\]\|\[Feature:.+\] --minStartupPods=8 --report-dir=/workspace/_artifacts --disable-log-dump=true: exit status 1
e2e.go: Test expand_less | 22m51s
error during ./hack/ginkgo-e2e.sh --ginkgo.skip=\[Slow\]\|\[Serial\]\|\[Disruptive\]\|\[Flaky\]\|\[Feature:.+\] --minStartupPods=8 --report-dir=/workspace/_artifacts --disable-log-dump=true: exit status 1

**Review by soltysh — APPROVED** [review]
This is good for now, but before we promote these endpoint/verb tests to conformance I'd like to see an effort which will results in API tests being similar to https://github.com/kubernetes/kubernetes/blob/a7fe0bf8e593decabba030f771ea8dbb34d22710/test/e2e/network/ingress.go#L918-L1167 I've started doing that for cronjobs and I'd love see this being implemented for all other apps endpoints as a single test too. 

/lgtm
/approve

**Comment by Riaankl** [comment]
/test pull-kubernetes-e2e-gce-ubuntu-containerd

Unrelated flake
```
• Failure [123.131 seconds]
[k8s.io] Pods
test/e2e/framework/framework.go:635
  should run through the lifecycle of Pods and PodStatus [Conformance] [It]
  test/e2e/framework/framework.go:640
  Feb 24 21:57:13.098: failed to see Pod pod-test in namespace pods-1725 running
      
  Unexpected error:
      <*errors.errorString | 0xc0001c4240>: {
          s: "timed out waiting for the condition",
      }
      timed out waiting for the condition
  occurred
```

**Comment by Riaankl** [comment]
/test pull-kubernetes-e2e-kind
Unrelated flake
```
• Failure [123.131 seconds]
[k8s.io] Pods
test/e2e/framework/framework.go:635
  should run through the lifecycle of Pods and PodStatus [Conformance] [It]
  test/e2e/framework/framework.go:640
  Feb 24 21:57:13.098: failed to see Pod pod-test in namespace pods-1725 running
      
  Unexpected error:
      <*errors.errorString | 0xc0001c4240>: {
          s: "timed out waiting for the condition",
      }
      timed out waiting for the condition
  occurred
```

**Comment by Riaankl** [comment]
/test pull-kubernetes-e2e-kind

Unrelated flake
```
• Failure [123.131 seconds]
[k8s.io] Pods
test/e2e/framework/framework.go:635
  should run through the lifecycle of Pods and PodStatus [Conformance] [It]
  test/e2e/framework/framework.go:640
  Feb 24 21:57:13.098: failed to see Pod pod-test in namespace pods-1725 running
      
  Unexpected error:
      <*errors.errorString | 0xc0001c4240>: {
          s: "timed out waiting for the condition",
      }
      timed out waiting for the condition
  occurred
```

**Comment by Riaankl** [comment]
/test pull-kubernetes-node-e2e

Unrelated flake
```
Detecting project
I0225 06:57:45.691] Project: k8s-infra-e2e-boskos-090
I0225 06:57:45.691] Network Project: k8s-infra-e2e-boskos-090
I0225 06:57:45.691] Zone: us-west1-b
I0225 06:57:45.691] Dumping logs from master locally to '/workspace/_artifacts'
W0225 06:57:46.326] ERROR: (gcloud.compute.addresses.describe) Could not fetch resource:
W0225 06:57:46.326]  - The resource 'projects/k8s-infra-e2e-boskos-090/regions/us-west1/addresses/bootstrap-e2e-master-ip' was not found
W0225 06:57:46.326] 
W0225 06:57:46.433] Could not detect Kubernetes master node.  Make sure you've launched a cluster with 'kube-up.sh'
I0225 06:57:46.533] Master not detected. Is the cluster up?
I0225 06:57:46.533] Dumping logs from nodes locally to '/workspace/_artifacts'
```

**Comment by Riaankl** [comment]
/test pull-kubernetes-node-e2e

Unrelated flake
```
I0225 08:16:35.853] Detecting project
I0225 08:16:35.853] Project: k8s-infra-e2e-boskos-035
I0225 08:16:35.853] Network Project: k8s-infra-e2e-boskos-035
I0225 08:16:35.854] Zone: us-west1-b
I0225 08:16:35.854] Dumping logs from master locally to '/workspace/_artifacts'
W0225 08:16:36.629] ERROR: (gcloud.compute.addresses.describe) Could not fetch resource:
W0225 08:16:36.629]  - The resource 'projects/k8s-infra-e2e-boskos-035/regions/us-west1/addresses/bootstrap-e2e-master-ip' was not found

E0225 08:16:41.007] Command failed
I0225 08:16:41.007] process 552 exited with code 1 after 1.2m
E0225 08:16:41.007] FAIL: pull-kubernetes-node-e2e
I0225 08:16:41.008] Call:  gcloud auth activate-service-account --key-file=/etc/service-account/service-account.json
W0225 08:16:41.567] Activated service account credentials for: [prow-build@k8s-infra-prow-build.iam.gserviceaccount.com]
I0225 08:16:41.657] process 5730 exited with code 0 after 0.0m
```

**Comment by Riaankl** [comment]
Unrelated flake - Will give some time and retest

```
I0225 08:42:58.644] Detecting project
I0225 08:42:58.644] Project: k8s-infra-e2e-boskos-085
I0225 08:42:58.645] Network Project: k8s-infra-e2e-boskos-085
I0225 08:42:58.645] Zone: us-west1-b
I0225 08:42:58.645] Dumping logs from master locally to '/workspace/_artifacts'
W0225 08:42:59.341] ERROR: (gcloud.compute.addresses.describe) Could not fetch resource:
W0225 08:42:59.342]  - The resource 'projects/k8s-infra-e2e-boskos-085/regions/us-west1/addresses/bootstrap-e2e-master-ip' was not found
W0225 08:42:59.342] 
W0225 08:42:59.483] Could not detect Kubernetes master node.  Make sure you've launched a cluster with 'kube-up.sh'
I0225 08:42:59.584] Master not detected. Is the cluster up?
I0225 08:42:59.584] Dumping logs from nodes locally to '/workspace/_artifacts'
```

**Comment by Riaankl** [comment]
Unrelated flake
```
W0225 11:10:10.010] ERROR: (gcloud.compute.addresses.describe) Could not fetch resource:
W0225 11:10:10.011]  - The resource 'projects/k8s-infra-e2e-boskos-009/regions/us-west1/addresses/bootstrap-e2e-master-ip' was not found
W0225 11:10:10.011] 
W0225 11:10:10.211] Could not detect Kubernetes master node.  Make sure you've launched a cluster with 'kube-up.sh'
I0225 11:10:10.312] Master not detected. Is the cluster up?
I0225 11:10:10.312] Dumping logs from nodes locally to '/workspace/_artifacts'
```

**Comment by Riaankl** [comment]
Unrelated flakes
```
Kubernetes e2e suite: [sig-apps] Deployment should run the lifecycle of a Deployment [Conformance] expand_more1m1s | Kubernetes e2e suite: [sig-apps] Deployment should run the lifecycle of a Deployment [Conformance] expand_more | 1m1s
Kubernetes e2e suite: [sig-apps] Deployment should run the lifecycle of a Deployment [Conformance] expand_more | 1m1s
Kubernetes e2e suite: [sig-network] KubeProxy should set TCP CLOSE_WAIT timeout [Privileged] expand_more4m46s | Kubernetes e2e suite: [sig-network] KubeProxy should set TCP CLOSE_WAIT timeout [Privileged] expand_more | 4m46s
Kubernetes e2e suite: [sig-network] KubeProxy should set TCP CLOSE_WAIT timeout [Privileged] expand_more | 4m46s
Kubernetes e2e suite: [sig-storage] PersistentVolumes-local [Volume type: blockfswithoutformat] Two pods mounting a local volume at the same time should be able to write from pod1 and read from pod2 expand_more | Kubernetes e2e suite: [sig-storage] PersistentVolumes-local [Volume type: blockfswithoutformat] Two pods mounting a local volume at the same time should be able to write from pod1 and read from pod2 expand_more
Kubernetes e2e suite: [sig-storage] PersistentVolumes-local [Volume type: blockfswithoutformat] Two pods mounting a local volume at the same time should be able to write from pod1 and read from pod2 expand_more
```

**Pull request merged**